src/pytesmo/validation_framework/adapters.py

```diff
@@ -225,53 +316,70 @@ def __mask(self, data):
         return data[mask]
 
     def read_ts(self, *args, **kwargs):
-        data = super(AdvancedMaskingAdapter, self).read_ts(*args, **kwargs)
+        data = super().read_ts(*args, **kwargs)
         return self.__mask(data)
 
     def read(self, *args, **kwargs):
-        data = super(AdvancedMaskingAdapter, self).read(*args, **kwargs)
+        data = super().read(*args, **kwargs)
         return self.__mask(data)
 
+    def _adapt_custom(self, *args, **kwargs):
+        data = super()._adapt_custom(*args, **kwargs)
+        return self.__mask(data)
+
+
 class AnomalyAdapter(BasicAdapter):
```

Comment:
Would it make sense to merge `AnomalyAdapter` and `AnomalyClimAdapter` to a single class and decide on the different behaviours based on a keyword argument? They are very similar.

Reply:
maybe, they require different arguments (`window_size` only vs many options for the climatology calculation). We can combine them for sure, but as we use them both in qa4sm I would do this in a separate PR.

Reply:
Yes, probably better